pgscatalog.utils/packages/pgscatalog.validate/src/pgscatalog/validate/lib/models.py

```
@@ -0,0 +1,212 @@
+
+        # allelefrequency_effect_[Ancestry] variable columns
+        pattern = r'^allelefrequency_effect_'
+        columns = {c for c in columns if not re.match(pattern, c)}
+
+        # Don't allow non-listed columns
+        allowed_columns = self.base_columns.union(self.hm_columns) if self.hm else self.base_columns
+        if not columns.issubset(allowed_columns):
+            raise ValueError(errors.UNEXPECTED_COLUMNS.format(columns=str(columns - set(allowed_columns))))
+
+        return self
+
+
+class ValidationVariant(ScoreVariant):
+    """Adds an extra layer of validation of scoring file variants"""
+
+    ## Allowed chromosome names
+    # Either 1..22, X, Y, XY or MT
+    allowed_chr_names: ClassVar[set[str]] = {
+        *map(str, range(1, 23)),
+        'X',
+        'Y',
+        'XY',
+        'MT'
+    }
+    # Or random/alt contigs like "1_KI270766v1_alt"
+    _contig_pattern: ClassVar[re.Pattern] = re.compile(
+        rf"^({'|'.join(map(re.escape, allowed_chr_names))})_([^_]+)_(alt|random)$"
+    )
+
+    @staticmethod
+    def __value_contains_non_printable_characters(value) -> bool:
+        return any([not c.isprintable() for c in str(value)])
+
+    @staticmethod
+    def __value_contains_leading_or_trailing_spaces(value) -> bool:
+        return str(value).strip() != str(value)
+
+    @field_validator(
+        *ColumnNames.base_columns, *ColumnNames.hm_columns,
+        mode="before",
+        check_fields=False,
+    )
+    @classmethod
+    def extra_field_validation(cls, v: Any) -> Optional[Any]:
+        """
+        Validate all fields for common mistakes such as non-printable characters or leading/trailing spaces.
+        """
+        if cls.__value_contains_non_printable_characters(v):
+            raise ValueError(errors.NON_PRINTABLE_CHAR.format(input=repr(v)))
+        if cls.__value_contains_leading_or_trailing_spaces(v):
+            raise ValueError(errors.LEADING_OR_TRAILING_SPACE.format(input=repr(v)))
```

do you prefer to throw an error here or [just trim the whitespace](https://docs.pydantic.dev/2.0/usage/types/string_types/#arguments-to-constr)?

---

Good point, a warning would be better here but there is no level of severity implemented at the moment. We might consider ignoring it for now and trim the whitespaces anyway during the import of the raw scores to PGS formatted scores.